### PR TITLE
UIC-787: heat map color generator

### PR DESCRIPTION
**Pull request by avantika-karwasara**
branch : dev/heat map color

- heat map color generation algorithm added using => min color, max color, min value, max value, and current value

### CODE REVIEW

File: superset/assets/src/visualizations/leaflet_map.js [hunk 1]
```diff
@@ -108,34 +108,34 @@ function leafletmap(slice, payload) {
         }
     }
 
+    function getRangeValue(val, max, min) {
+      if(max - min === 0) {
+        return 1;
+      }
+      return (val - min) / (max - min);
+    }
```
Comment (arpit-agarwal) [inline superset/assets/src/visualizations/leaflet_map.js]: We need to take care of negative overflow when val > max  and val < min

File: superset/assets/src/visualizations/leaflet_map.js [hunk 2]
```diff
@@ -108,34 +108,43 @@ function leafletmap(slice, payload) {
...
+    function colourGradientor(rgb_beginning,rgb_end,p,max,min){
+        var rangeValue = getRangeValue(p,parseInt(max),parseInt(min));
+        var startWeight = rangeValue;
+        var endWeight = 1 - rangeValue;
+        var rgb = [parseInt(rgb_beginning.r * startWeight + rgb_end.r* endWeight),
+            parseInt(rgb_beginning.g * startWeight + rgb_end.g* endWeight),
+                parseInt(rgb_beginning.b * startWeight + rgb_end.b * endWeight),
+                parseInt(rgb_beginning.a * startWeight + rgb_end.a * endWeight)];
+        return 'rgb('+rgb[0] +',' + rgb[1] +',' +rgb[2] +','+rgb_beginning.a + ')';
+    }
```
[the remark this answers is not in the document]
Comment (arpit-agarwal) [reply superset/assets/src/visualizations/leaflet_map.js]: screenshot 2018-12-18 at 8 59 19 am

[the remark this answers is not in the document]
Comment (arpit-agarwal) [reply superset/assets/src/visualizations/leaflet_map.js]: http://www.andrewnoske.com/wiki/Code_-_heatmaps_and_color_gradients

File: superset/assets/src/visualizations/leaflet_map.js (hunk 1 above)
Comment (avantika-karwasara) [reply superset/assets/src/visualizations/leaflet_map.js]: changes done

File: superset/assets/src/visualizations/leaflet_map.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (avantika-karwasara) [reply superset/assets/src/visualizations/leaflet_map.js]: changes done

**Pull request merged**